xcp_d/utils/plot.py

```diff
@@ -1006,3 +1006,135 @@ def _get_tr(img):
+        # Iterates over volumetric and surface structures
+        if name == surf_name:  # Just looking for a surface
+            data = data.T[data_indices]
+            # Assume brainmodels axis is last, move it to front
+            vtx_indices = model.vertex
+            # Generally 1-N, except medial wall vertices
+            surf_data = np.zeros((vtx_indices.max() + 1,) + data.shape[1:], dtype=data.dtype)
+            surf_data[vtx_indices] = data
+            return surf_data
+
+    raise ValueError(f"No structure named {surf_name}")
+
+
+def plot_alff_reho_surface(output_path, filename, bold_file):
+    """Plot ReHo and ALFF for ciftis on surface."""
+    import matplotlib.pyplot as plt
+    import nibabel as nb
+    from bids.layout import parse_file_entities
+    from nilearn import plotting as plott
+    from templateflow.api import get as get_template
+
+    from xcp_d.utils.plot import surf_data_from_cifti
+
+    func = filename
+    try:
+        density = str(parse_file_entities(bold_file)["den"])
+    except Exception:
+        density = None
+
+    if density is None:
+        density = "32k"
+    rh = str(
+        get_template(template="fsLR", hemi="L", density=density, suffix="midthickness")
+    )
+    lh = str(
+        get_template(template="fsLR", hemi="R", density=density, suffix="midthickness")
+    )
+    cifti = nb.load(func)
+    cifti_data = cifti.get_fdata()
+    cifti_axes = [cifti.header.get_axis(i) for i in range(cifti.ndim)]
+
+    fig, axes = plt.subplots(figsize=(16, 16), ncols=2, nrows=2, subplot_kw={'projection': '3d'})
+
+    lh_surf_data = surf_data_from_cifti(
+        cifti_data,
+        cifti_axes[1],
+        'CIFTI_STRUCTURE_CORTEX_LEFT',
+    )
+    rh_surf_data = surf_data_from_cifti(
+        cifti_data,
+        cifti_axes[1],
+        'CIFTI_STRUCTURE_CORTEX_RIGHT',
+    )
+    plott.plot_surf_stat_map(
+        lh,
+        lh_surf_data,
+        hemi="left",
+        view="lateral",
+        engine="matplotlib",
+        colorbar=False,
+        axes=axes[0, 0],
+        figure=fig,
+    )
+    plott.plot_surf_stat_map(
+        lh,
+        lh_surf_data,
+        hemi="left",
+        view="medial",
+        engine="matplotlib",
+        colorbar=False,
+        axes=axes[1, 0],
+        figure=fig,
+    )
+    plott.plot_surf_stat_map(
+        rh,
+        rh_surf_data,
+        hemi="right",
+        view="lateral",
+        engine="matplotlib",
+        colorbar=False,
+        axes=axes[0, 1],
+        figure=fig,
+    )
+    plott.plot_surf_stat_map(
+        rh,
+        rh_surf_data,
+        hemi="right",
+        view="medial",
+        engine="matplotlib",
+        colorbar=False,
+        axes=axes[1, 1],
+        figure=fig,
+    )
+    axes[0, 0].set_title("Left Hemisphere", fontsize=40)
+    axes[0, 1].set_title("Right Hemisphere", fontsize=40)
+    fig.tight_layout()
+    fig.savefig(output_path)
+    output_pathname = output_path
+    return output_pathname
```

Comment:
You don't need to duplicate a variable like this.

```suggestion
    return output_path
```